packages/react-instantsearch-hooks-dom/src/types/OptionalKeys.ts

```diff
@@ -0,0 +1,3 @@
+export type OptionalKeys<T> = {
+  [K in keyof T]-?: {} extends Pick<T, K> ? K : never;
+}[keyof T];
```

Comment:
```suggestion
export type OptionalKeys<TObject> = {
  [TKey in keyof TObject]-?: {} extends Pick<TObject, TKey> ? TKey : never;
}[keyof TObject];
```

Reply:
I thought we had an eslint rule that banned single-letter variables, surprised this wasn't flagged

Reply:
Just did a check and could not find a rule for that. [`id-length`](https://github.com/algolia/eslint-config-algolia/blob/master/packages/eslint-config-algolia/rules/base.js#L184) is turned off by default and does not account for type parameters anyway. I can have that behavior checked with a custom regex in `@typescript-eslint/naming-convention`:

```js
      {
        selector: 'typeParameter',
        format: ['PascalCase'],
        // prefix: ['T', 'K'],
        custom: {
          regex: '^[T|K].+',
          match: true,
        },
      },
```

Wdyt?